server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -109,8 +109,23 @@ public TransportFieldCapabilitiesAction(
 
     @Override
     protected void doExecute(Task task, FieldCapabilitiesRequest request, final ActionListener<FieldCapabilitiesResponse> listener) {
+        ActionListener<FieldCapabilitiesResponse> loggingListener = listener.delegateResponse((l, e) -> {
+            int statusCode = ExceptionsHelper.status(e).getStatus();
+            if (statusCode >= 500) {
+                logger.debug(
+                    "{} Exception in TransportFieldCapabilitiesAction for task: [{}] and request: [{}]. Exception: {}. Stack trace: {}",
+                    statusCode,
+                    task,
+                    request,
+                    e.getMessage(),
+                    ExceptionsHelper.stackTrace(e)
+                );
+            }
```

Comment:
Typically you'd want to pass the exception to the logger directly rather than formatting it as a string yourself - that way it goes into the right fields in the JSON output.

```suggestion
                logger.debug(
                    Strings.format(
                        "%d Exception in TransportFieldCapabilitiesAction for task: [%s] and request: [%s].",
                        statusCode,
                        task,
                        request
                    ),
                    e
                );
```